src/dir/glob.cr

```diff
@@ -161,11 +161,15 @@ class Dir
           path_stack << {next_pos, root, nil}
         when DirectoriesOnly
           raise "unreachable" unless path
-          fullpath = path == File::SEPARATOR_STRING ? path : path + File::SEPARATOR
-          if dir_entry
-            yield fullpath if dir_entry.dir?
+          # FIXME: [win32] File::SEPARATOR_STRING comparison is not sufficient for Windows paths.
+          if path == File::SEPARATOR_STRING
+            fullpath = path
           else
-            yield fullpath if dir?(fullpath)
+            fullpath = Path[path].join("").to_s
+          end
+
+          if (dir_entry && dir_entry.dir?) || dir?(fullpath)
+            yield fullpath
```

Comment:
You managed to drop your fixup changes.

Never attempt to fixup another fixup commit, it breaks. Always re-fixup the original one

Reply:
Na, I just messed up the rebase. Did only pay atttention to the first change.